test/Microsoft.OpenAPI.OData.Reader.Tests/Edm/ODataPathProviderTests.cs

```diff
@@ -40,15 +40,18 @@ public void GetPathsForGraphBetaModelReturnsAllPaths()
         {
             // Arrange
             IEdmModel model = EdmModelHelper.GraphBetaModel;
-            var settings = new OpenApiConvertSettings();
+            var settings = new OpenApiConvertSettings()
+            {
+                ExpandDerivedTypesNavigationProperties = false
+            };
             ODataPathProvider provider = new ODataPathProvider();
 
             // Act
             var paths = provider.GetPaths(model, settings);
 
             // Assert
             Assert.NotNull(paths);
-            Assert.Equal(12261, paths.Count());
+            Assert.Equal(14624, paths.Count());
```

Comment:
since you disable the setting, why do you get more paths?

Reply:
The extra number of paths are because we are now generating OData type cast paths for collection-valued navigation properties that were initially missing, e.g.:

![image](https://user-images.githubusercontent.com/40403681/165133545-32bfdff4-80f7-415c-8291-7b9edd96417c.png)

What I've turned off in the setting by default is the expansion of OData type cast paths into their navigation properties, for single-valued nav. props. Expanding into those end up adding an absurd number of paths.